test/system/src/test/java/io/pravega/test/system/DynamicRestApiTest.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    /**
+     * This is used to setup the various services required by the system test framework.
+     *
+     * @throws InterruptedException If interrupted
+     * @throws MarathonException    when error in setup
+     * @throws URISyntaxException   If URI is invalid
+     */
+    @Environment
+    public static void initialize() throws MarathonException {
+        URI zkUri = startZookeeperInstance();
+        startBookkeeperInstances(zkUri);
+        URI controllerUri = ensureControllerRunning(zkUri);
+        ensureSegmentStoreRunning(zkUri, controllerUri);
+    }
+
+    @Test
+    public void listScopes() {
+        Service conService = Utils.createPravegaControllerService(null);
+        List<URI> ctlURIs = conService.getServiceDetails();
+        URI controllerRESTUri = ctlURIs.get(1);
+        Invocation.Builder builder;
+        Response response;
+        String protocol = Utils.TLS_AND_AUTH_ENABLED ? "https://" : "http://";
+        restServerURI = protocol + controllerRESTUri.getHost() + ":" + controllerRESTUri.getPort();
+        log.info("REST Server URI: {}", restServerURI);
+
+        assertFalse(true);
+
+        // TEST REST server status, ping test
+        resourceURl = new StringBuilder(restServerURI).append("/ping").toString();
+        webTarget = client.target(resourceURl);
+        builder = webTarget.request();
+        response = builder.get();
+        assertEquals("Ping test", OK.getStatusCode(), response.getStatus());
+        log.info("REST Server is running. Ping successful.");
+
+        final String scope1 = RandomStringUtils.randomAlphanumeric(10);
+        final String stream1 = RandomStringUtils.randomAlphanumeric(10);
+
+        Client client = ClientBuilder.newClient();
+        String responseAsString = null;
+
+        ClientConfig clientConfig = ClientConfig.builder()
+                        .controllerURI(controllerRESTUri)
+                        .credentials(new DefaultCredentials("1111_aaaa", "admin"))
+                        .build();
+        // Create a scope.
+        @Cleanup
+        StreamManager streamManager = StreamManager.create(clientConfig);
+        assertNotNull(streamManager);
+        boolean isScopeCreated = streamManager.createScope(scope1);
+        assertTrue("Failed to create scope", isScopeCreated);
+        // Create a stream.
+        boolean isStreamCreated = streamManager.createStream(scope1, stream1, StreamConfiguration.builder().scalingPolicy(ScalingPolicy.fixed(1)).build());
+        assertTrue("Failed to create stream", isScopeCreated);
+
+        // Query for an existing scope.
+        webTarget = client.target(restServerURI).path("v1").path("scopes");
+        builder = webTarget.request();
+        response = builder.get();
+        assertEquals("Get scopes failed.", OK.getStatusCode(), response.getStatus());
+        responseAsString = response.readEntity(String.class);
+        assertTrue(responseAsString.contains("\"scopeName\":\"_system\""));
```

Comment:
Since you create a scope and stream earlier, you should also check if that shows up in the response too? Otherwise creating the scope/stream previously adds no value in this particular test.

Reply:
Sure.